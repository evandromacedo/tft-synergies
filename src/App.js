import React, { useState, useEffect } from 'react';
import logo from './logo.svg';
import './App.css';

function App() {
  const [firstCount, updateFirstCount] = useState(0);
  const [secondCount, updateSecondCount] = useState(100);

  function handleFirstCounter() {
    if (firstCount + 1 > 10) {
      return updateFirstCount(0);
    }

    updateFirstCount(firstCount + 1);
  }

  return (
    <div className="App">
      <div>
        <p>First counter: {firstCount}</p>
        <button onClick={handleFirstCounter}>Update First Counter</button>
        <p>Second counter: {secondCount}</p>
        <button onClick={() => updateSecondCount(secondCount + 1)}>
          Update Second Counter
        </button>
      </div>

      {/* <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header> */}
    </div>
  );
}

export default App;
