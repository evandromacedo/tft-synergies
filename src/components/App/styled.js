import { createGlobalStyle } from 'styled-components';
import BeaufortRegular from '../../static/font/Beaufort-Regular.ttf';
import BeaufortBold from '../../static/font/Beaufort-Bold.ttf';

export const GlobalStyle = createGlobalStyle`
  /* Resets */
  *, *:before, *:after {
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
  }

  html, body, div, span, applet, object, iframe,
  h1, h2, h3, h4, h5, h6, p, blockquote, pre,
  a, abbr, acronym, address, big, cite, code,
  del, dfn, em, img, ins, kbd, q, s, samp,
  small, strike, strong, sub, sup, tt, var,
  b, u, i, center,
  dl, dt, dd, ol, ul, li,
  fieldset, form, label, legend,
  table, caption, tbody, tfoot, thead, tr, th, td,
  article, aside, canvas, details, embed,
  figure, figcaption, footer, header, hgroup,
  menu, nav, output, ruby, section, summary,
  time, mark, audio, video {
    margin: 0;
    padding: 0;
    border: 0;
    vertical-align: baseline;
  }

  body {
    -webkit-tap-highlight-color: transparent;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    line-height: 1;
  }

  ol, ul {
    list-style: none;
  }

  a {
    color: inherit;
    text-decoration: inherit;
  }

  input {
    outline-width: 0.3em;
    outline-color: transparent;
    -webkit-appearance: none;
    appearance: none;
    -webkit-border-radius:0;
    border-radius:0;
  }

  .input:disabled {
    background-color: transparent;
  }

  button {
    outline: none;
    cursor: pointer;
    background-color: transparent;
    margin: 0;
  }

  input,
  button {
    text-transform: inherit;
    font-family: inherit;
    letter-spacing: inherit;
    border: none;
  }

  /* Responsiveness */
  img,
  iframe,
  object,
  embed,
  video {
    height: auto;
    max-width: 100%;
  }

  /* Fonts */
  @font-face {
    font-family: 'Beaufort';
    src: url(${BeaufortRegular}) format('truetype');
    font-weight: normal;
    font-style: normal;
  }

  @font-face {
    font-family: 'Beaufort';
    src: url(${BeaufortBold}) format('truetype');
    font-weight: bold;
    font-style: normal;
  }

  html {
    font-family: 'Beaufort', sans-serif;
    font-weight: bold;
    font-size: 10px;
    color: #fff;
    background-color: #121212;
  }
`;
