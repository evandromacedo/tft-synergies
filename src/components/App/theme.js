const theme = {
  // Colors
  background1: '#000000',
  background2: '#121212',
  background3: '#343434',
  background4: '#979797',
  accentYellow: '#D8A75B',
  borderTier1: {
    top: '#213042',
    bottom: '#131C26'
  },
  borderTier2: {
    top: '#156831',
    bottom: '#0A3518'
  },
  borderTier3: {
    top: '#1652A0',
    bottom: '#092242'
  },
  borderTier4: {
    top: '#893088',
    bottom: '#3E143D'
  },
  borderTier5: {
    top: '#B89D27',
    bottom: '#3A310B'
  },

  // Typography
  bold12px: 'font-size: 1.2rem; font-weight: bold; letter-spacing: .5px;',
  bold16px: 'font-size: 1.6rem; font-weight: bold; letter-spacing: .5px;',
  bold20px: 'font-size: 2rem; font-weight: bold; letter-spacing: .5px;',
  sans14px: 'font-family: sans-serif; font-size: 1.4rem; font-weight: normal;',
  sans14pxItalic:
    'font-family: sans-serif; font-size: 1.4rem; font-weight: normal; font-style: italic;',
  sans16pxItalic:
    'font-family: sans-serif; font-size: 1.6rem; font-weight: normal; font-style: italic;',
  textShadow1: 'text-shadow: 0 0 3px #000;',

  // Breakpoints
  large: '1024px',
  medium: '768px',
  small: '480px'
};

export default theme;
