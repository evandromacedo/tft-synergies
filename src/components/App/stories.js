import React from 'react';

import App from './index.js';

import { storiesOf } from '@storybook/react';

storiesOf('App', module).add('WTFFF', () => <App />);
