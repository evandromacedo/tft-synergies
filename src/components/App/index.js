import React from 'react';
import { ThemeProvider } from 'styled-components';
import { Container, Inner, GlobalStyle } from './styled';
import theme from './theme';
import Meta from '../Meta';
import Header from '../Header';

export default function App() {
  return (
    <>
      <GlobalStyle />
      <ThemeProvider theme={theme}>
        <Container>
          <Meta />
          <Header />
          <Inner></Inner>
        </Container>
      </ThemeProvider>
    </>
  );
}
