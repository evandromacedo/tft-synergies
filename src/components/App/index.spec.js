import React from 'react';
import { shallow } from 'enzyme';
import App from '.';

it('renders without crashing', () => {
  // shallow(<App />);
  expect(1).toEqual(1);
});
