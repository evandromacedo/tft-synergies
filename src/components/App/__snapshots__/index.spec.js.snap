// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders the Container 1`] = `
<styled.div>
  <Alo />
</styled.div>
`;
